insights/specs.py

```diff
@@ -34,8 +34,35 @@
 brctl_show = sf.simple_command("/usr/sbin/brctl show", name="brctl_show")
 candlepin_log = sf.simple_file("/var/log/candlepin/candlepin.log", name="candlepin_log", alias="candlepin.log")
 candlepin_error_log = sf.first_of([sf.simple_command("/var/log/candlepin/error.log"), sf.simple_file(r"sos_commands/foreman/foreman-debug/var/log/candlepin/error.log", context=HostArchiveContext)], name="candlepin_error_log")
-catalina_out = sf.glob_file(["/var/log/tomcat/catalina.out", "/var/log/tomcat6/catalina.out", "/tomcat-logs/tomcat/catalina.out", "/tomcat-logs/tomcat6/catalina.out"], name="catalina_out", alias="catalina.out")
-catalina_server_log = sf.glob_file(["/var/log/tomcat/logs/catalina*.log", "/var/log/tomcat6/logs/catalina*.log", "/tomcat-logs/tomcat/catalina*.log", "/tomcat-logs/tomcat6/catalina*.log"], name="catalina_server_log")
+ps_auxww = sf.first_of([
+    sf.simple_command("/bin/ps auxww", name="ps_auxww"),
+    sf.simple_file('sos_commands/process/ps_aux', alias='ps_aux', context=HostArchiveContext),
+    sf.simple_file('sos_commands/process/ps_auxwww', alias='ps_auxwww', context=HostArchiveContext),
+    sf.simple_file('sos_commands/process/ps_auxcww', alias='ps_auxcww', context=HostArchiveContext),
+])
+
+
+@datasource(ps_auxww)
+def tomcat_base(broker):
+    ps = broker[ps_auxww].content
+    results = []
+    findall = re.compile(r"\-Dcatalina\.base=(\S+)").findall
+    for p in ps:
+        found = findall(p)
+        if found:
+            # Only get the path which is absolute
+            results.extend(f for f in found if f[0] == '/')
+    return list(set(results))
+
+
+catalina_out = sf.first_of([sf.foreach(tomcat_base, "%s/catalina.out"),
```

Comment:
`catalina.out` should locate in $tomcat_base/logs/

Reply:
Thanks @zhangyi733, I updated it in the new commit.